### ISSUE TITLE
Vectorized matmul performance regression - function inlining

### ISSUE THREAD

**Issue by jtuyls** [nod-ai/iree-amd-aie#883]
We're seeing performance regression on vectorized matmul, likely caused by the following PR: https://github.com/nod-ai/iree-amd-aie/pull/856, see table below:

Matmul problem size: 512x512x4096 (MxKxN)
Array configuration: 2x2
Vectorization or ukernel or scalar: Vectorization 

| Commit | Latency (us) |
| -------- | ------- |
| 12f0502 | 48521 |
| 2086718 | 42513 |

@Abhishek-Varma 

Note that there is another PR causing performance regression: https://github.com/nod-ai/iree-amd-aie/issues/882, which is likely orthogonal.

**Comment by newling** [comment]
Can someone please remind me -- at what granularity is the matmul outlined? Is it at the m=n=k=64 granularity or the m=n=4 k=8 granularity (assuming phoenix bf16) ?

**Comment by Abhishek-Varma** [comment]
Not sure, but perhaps [this](https://github.com/nod-ai/iree-amd-aie/pull/888) might be the reason behind (and hopefully a fix for) this regression.

**Comment by newling** [comment]
Maybe, but it isn't surprising to me that outlining a single AIE instruction (matmul on 4x8x4) can result in a slow down

**Comment by Abhishek-Varma** [comment]
> Maybe, but it isn't surprising to me that outlining a single AIE instruction (matmul on 4x8x4) can result in a slow down

Yeah, I guess outlining functions would definitely add some regression because of function invocation overhead.

As it was initially attempted to reduce the program memory requirement, it can definitely introduce performance overhead - perhaps the way forward should be "conditional" enabling of function outlining for now while the peano loop unrolling control is enabled ?
